### DISCUSSION TITLE
Cannot Generate Molecule From MolFile - No Examples?

### DISCUSSION THREAD

**Discussion by Eelesy** [rdkit/rdkit-js#358] [Q&A] [answered]
Hi,

I can generate molecules using 'get_mol()' and SMILES quite easily.

I have a need to generate via MolFiles or MolBlocks - the documentation says it takes them in as an input but I am tearing my hair out trying to get one to work?

I have the files stored locally, I assume a file path is not right? I have used a http client in Angualr to load the content of the file as a string and pass that, no luck. I have also just created the contents of the mol file as a var and passed that in and that does not work.

Has anyone had any joy with MolFiles?

**Comment by ptosco** [comment, marked as answer]
`get_mol` will accept single molecules as either SMILES strings or MDL molblock strings. File paths won't work, nor will SDF files (i.e., concatenations of individual molblocks with optional properties).
Here you find an example:
https://github.com/rdkit/rdkit/blob/409947314f9f53aded4dfab0697f2b720f347387/Code/MinimalLib/tests/tests.js#L204

**Reply by Eelesy** [reply]
Thank! My formatting was off. I will have to read files in as strings and pass over to the function. Thank you
